src/gui/screen_menu_esp_update.cpp

```diff
@@ -128,6 +132,28 @@ ScreenFactory::UniquePtr GetScreenMenuESPUpdate() {
     return ScreenFactory::Screen<ScreenMenuESPUpdate>();
 }
 
+void ScreenMenuESPUpdate::updateProgress() {
+    // Assumption: The file name always contains at least one slash, they come from this file.
+    const char *basename = rindex(current_file->filename, '/') + 1;
+    // Overestimating the buffer size a bit.
+    const size_t len = strlen(basename) + 20;
+    char *buffer = new char[len];
+    snprintf(buffer, len, "%s\n%" PRIu32 "/%" PRIu32, basename, readCount, current_file->size);
+    /*
+     * FIXME: Is there a better way?
+     *
+     * We apparently need to keep the source of the text around, not have it on stack.
+     * We also can't reuse the buffer between calls because it
+     * doesn't redraw (the SetText checks for _pointer_
+     * equality).
+     *
+     * Is there a way to get rid of that "translation" here?
+     */
+    help.SetText(_(buffer));
```

Comment:
use `stringview::MAKERAM` instead `_()`
`_()` would try to translate text, which cannot be done with dynamical texts

Also if you use `std::array` than you must manually call `help.Invalidate()`, because text will have same address, in this case `help.SetText` call should be only in ctor